pkg/vulnsrc/glad/glad.go

```diff
@@ -21,20 +21,21 @@ const (
 	// GitLab Advisory Database
 	gladDir = "glad"
 
-	Conan     packageType = "Conan"
-	Gem       packageType = "Gem"
-	Go        packageType = "Go"
-	Maven     packageType = "Maven"
-	Npm       packageType = "Npm"
-	Nuget     packageType = "Nuget"
-	Packagist packageType = "Packagist"
-	PyPI      packageType = "PyPI"
+	Conan packageType = "Conan"
+	Maven packageType = "Maven"
 )
 
 var (
-	// TODO: support Npm, NuGet, PyPI and Packagist
-	supportedPkgTypes   = []packageType{Maven, Conan}
-	supportedIDPrefixes = []string{"CVE", "GHSA", "GMS"}
+	supportedIDPrefixes = []string{
+		"CVE",
+		"GHSA",
+		"GMS",
+	}
+
+	ecosystems = map[packageType]types.Ecosystem{
+		Maven: vulnerability.Maven,
+		Conan: vulnerability.Conan,
+	}
```

Comment:
What if we just use slice of ecosystems? It should avoid confusion between ecosystems and packageTypes.
```suggestion
	ecosystems = []types.Ecosystem{
		vulnerability.Maven,
		vulnerability.Conan,
	} 
```

Reply:
They can be different values because types.Ecosystem is defined by us. We anyway need a mapping between GitLab package types and our ecosystem values, right? Do you have any ideas?

Reply:
When i checked code - i found that we only use `packageType` for path to advisories.
But after your comment i understand that we use glad types in `vuln-list-update` - https://github.com/aquasecurity/vuln-list-update/blob/26dae1a5f7eccb8091db8288b679d215a8b11d88/glad/glad.go#L163
And if we remove `packageType` - we can get error in case, when ecosystem name != GitLab package types (as is case with Rust and Cargo for ghsa).

Now i think your solution is okay.
But can you add link to GitLab package types - https://gitlab.com/gitlab-org/security-products/gemnasium-db#package-slug-and-package-name

Reply:
Good idea. Pushed.
https://github.com/aquasecurity/trivy-db/pull/358/commits/a73bf41682e2f44e2ed9c038b616d80642a42bc6